libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -562,14 +584,19 @@ private async Task<bool> QueueFirstMatchAsync(SequenceContext sequenceContext, D
             return false;
         }
 
-        private void EnsureDependenciesInstalled()
+        private void EnsureDependenciesInstalled(DialogContext dc)
```

Comment:
>EnsureDependenciesInstalled [](start = 21, length = 27)

Can we rename this or split it apart? I find it confusing that you're using it to both setup your selector and initialize your dialog state.  I'm not even sure we need this.

We should be able to setup the selector on first access via a getter and the dialog state should be initialized as part of begin dialog.

Reply:
Changed this to do it on BeginDialog now.